### DISCUSSION TITLE
Guidance Migrating CRA to NX in stages

### DISCUSSION THREAD

**Discussion by raywong-fb** [nrwl/nx#15809] [General]
As the title implies, I'm in the process of migrating a react app generated by CRA.

In thinking about how I want to approach this, there were a few options I considered.
- Migrate to a monorepo setup
- Switch to a micro-frontend architecture
- Switch to Nx, but keep a standalone react app

Obviously, the first two options seem to be a much higher level of effort, something I could not take the time to do all at once. Potentially, creating micro-frontends would allow me to chunk out pieces of the legacy app and replace it w/ the micro-frontend little by little.

Then I thought about using the `cra-to-nx` migration script to "quickly" migrate off of CRA onto the Nx ecosystem. Once I'm on Nx as a standalone react app, does that make it any easier to then transition to a monorepo setup? In other words, is it worth it to migrate off of CRA keeping the standalone app structure, and then potentially move to a monorepo down the road? Or, if I know that I ultimately want to go to a monorepo is it better to do that all at once when I'm ready?

Would love to hear your feedback and/or personal experience you have that might be related. Thanks in advance.

**Comment by amartincolby** [comment]
What decisions did you land on and why? I am not struggling with CRA transition, but I am trying to understand the benefits of the various possible decisions.

**Reply by raywong-fb** [reply]
I ended up migrating to a standalone app.

As an organization we haven't decided if moving to a full monorepo is our long term goal, so that was obviously a big factor in the decision. Another factor was considering the effort to migrate. We didn't want to add additional complexity at this time (therefore extending the timeline) as we already anticipated time to switch from webpack to vite, jest to vitest, and upgrading to latest React and typescript versions. I also didn't want to have to modify the CI configuration too much, so switching to a standalone app was pretty seamless.
